[Security Solution] persists the alert details flyout overview panel expandable section open/collapsed state in localStorage

PhilippeOberti opened the pull request:
## Summary

This PR is the first of a few that are adding some UI persistence to the alert details expandable flyout to improve usability.

This one focuses on saving the expanded/collapsed state of each section (`About`, `Investigation`, `Visualizations`, `Insights` and `Response`) of the `Overview` tab only. This will allow users to navigate between alerts in the alerts table in the backend, open new flyouts and see their layout retained.

The value store in local storage it pretty basic:
```
{
  about: true/false,
  investigation: true/false,
  visualizations: true/false,
  insights: true/false,
  response: true/false,
}
```

Each section still have their default value (all collapsed except for `About` and `Investigation` which are expanded by default). The values saved in localStorage will override this default value.

https://github.com/elastic/kibana/assets/17276605/7b253ab4-0103-4a23-9c66-b5d9cdc18720

https://github.com/elastic/security-team/issues/7670

#### Notes:
_We had a discussion during standup to go through the Redux store and have a middleware handle the storing in localStorage. I decided to not apply this logic at the moment for the following reasons:_
- _the code was already finished, working and fully unit tested_
- _that would increase the size and complexity of this PR_
- _we do not have currently anything related to the flyout panels saved in localStorage and I felt this small feature was not the best candidate to get started_
- _this can be added later when we have a clearer picture on how much we'll be handling, with no impact on users as the value stored in localStorage should remain the same_
_Let me know if you disagree with these thoughts and I can revisit the approach!_

### Checklist

- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

PhilippeOberti committed:
[Security Solution] persists the alert details flyout overview panel expandable section open/collapsed state in localStorage

elasticmachine commented:
Pinging @elastic/security-threat-hunting-investigations (Team:Threat Hunting:Investigations)

christineweng reviewed (approved):
@PhilippeOberti This is a great addition to further improve user experience! Desk tested and works well. Just had a nit comment about making the local storage hook more general

PhilippeOberti committed:
use storage directly within the hooks

PhilippeOberti commented on x-pack/plugins/security_solution/public/flyout/document_details/right/hooks/use_expand_section.ts:
```diff
@@ -0,0 +1,40 @@
+/**
+ * Hook to get the expanded state of a section from local storage.
+ */
+export const useExpandSection = ({
```
yup I'll keep that in mind as I'm already working on the next PR that adds more local storage interaction!

There is actually already a `useLocalStorage` [hook](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/common/components/local_storage/index.tsx) that Andrew wrote but it is too restrictive. It does not allow me to have a nested object in localStorage and only update one key/value pair in that object...

PhilippeOberti commented:
@christineweng I simplified things a bit in my [last commit](https://github.com/elastic/kibana/pull/178746/commits/8910a98774994e66b38057062a6cbc59e37782af). I didn't like that I had to pass down the Kibana's `storage` object down to the hooks so I cleaned that up. It makes a lot more sense to me now.

Can you briefly check the changes and let me know if you like them as well?

PhilippeOberti commented:
need [this PR](https://github.com/elastic/kibana/pull/179469) to be merged first
